src/app/static/src/app/store/genericChart/actions.ts

```diff
@@ -40,5 +42,26 @@ export const actions: ActionTree<GenericChartState, RootState> & MetadataActions
                     });
                 }
             });
+    },
+    async refreshDatasets(context){
+        const {dispatch, state} = context;
+        if (!state.genericChartMetadata) {
+            return;
+        }
+
+        const datasetUrls = Object.values(state.genericChartMetadata)
+                            .reduce((urls: Dict<string>, chart: GenericChartMetadata) => {
+                                chart.datasets.forEach((dataset: DatasetConfig) => {
+                                    urls[dataset.id] = dataset.url
+                                });
+                                return urls;
+                            }, {});
+
+        const getDatasetActions: Promise<unknown>[] = [];
+        Object.keys(state.datasets).forEach((datasetId: string) => {
+            const url = datasetUrls[datasetId];
+            getDatasetActions.push(dispatch("getDataset", {datasetId, url}));
+        });
```

Comment:
I would find something like this easier to read, but very subjective:
```suggestion
        const getDatasetActions = Object.keys(state.datasets).map(datasetId => dispatch("getDataset", {datasetId, url: datasetUrls[datasetId]}));
```